velox/docs/functions/spark/string.rst

```diff
@@ -0,0 +1,87 @@
+====================================
+String Functions
+====================================
+
+.. note::
+
+    These functions assume that the input strings contain valid UTF-8 encoded
+    Unicode code points. There are no explicit checks for valid UTF-8 and
+    the functions may return incorrect results on invalid UTF-8.
+
+    Additionally, the functions operate on Unicode code points and not user
+    visible *characters* (or *grapheme clusters*).  Some languages combine
+    multiple code points into a single user-perceived *character*, the basic
+    unit of a writing system for a language, but the functions will treat each
+    code point as a separate unit.
+
+.. spark:function:: ascii(string) -> integer
+
+    Returns the numeric value of the first character of ``string``.
+
+.. spark:function:: chr(n) -> varchar
+
+    Returns the Unicode code point ``n`` as a single character string.
+
+.. spark:function:: contains(left, right) -> boolean
+
+    Returns a boolean. The value is True if right is found inside left.
+    Returns NULL if either input expression is NULL. Otherwise, returns False.
+    Both left or right must be of STRING. ::
+
+        SELECT contains('js SQL', 'js'); -- true
+        SELECT contains('js SQL', 'js'); -- false
+        SELECT contains('js SQL', null); -- NULL
+
+.. spark:function:: endsWith(left, right) -> boolean
+
+    Returns a boolean. The value is True if left ends with right.
+    Returns NULL if either input expression is NULL. Otherwise, returns False.
+    Both left or right must be of STRING. ::
+
+        SELECT endswith('js SQL', 'SQL'); -- true
+        SELECT endswith('js SQL', 'js'); -- false
+        SELECT endswith('js SQL', null); -- NULL
+
+.. spark:function:: instr(string, substring) -> integer
+
+    Returns the starting position of the first instance of ``substring`` in
+    ``string``. Positions start with ``1``. If not found, ``0`` is returned.
+
+.. spark:function:: length(string) -> integer
+
+    Returns the length of ``string`` in characters.
+
+.. spark:function:: split(string, delimiter) -> array(string)
+
+    Splits ``string`` on ``delimiter`` and returns an array. ::
```

Comment:
Perhaps, clarify what happens if delimiter is not found or an empty string.

Reply:
Added example：
        SELECT split('one', ''); -- ["o", "n", "e", ""]
        SELECT split('one', '1'); -- ["one"]